### ISSUE TITLE
Marketplace Tutorial: Transaction1 (Setting up the Sale) fails with GraphQL error

### ISSUE THREAD

**Issue by yellowBirdy** [onflow/flow-playground#45]
**Describe the bug**
Executing Transaction1.cdc (signed by acc 0x01) results in the playground hanging and the following error is logged in the console: 

```
> vendors~main.b46266fa.js:1 Uncaught (in promise) Error: GraphQL error: failed to parse events: failed to convert event value: cannot convert value of type interpreter.UFix64Value
    at new t (vendors~main.b46266fa.js:1)
    at Object.next (vendors~main.b46266fa.js:1)
    at _ (vendors~main.b46266fa.js:1)
    at w (vendors~main.b46266fa.js:1)
    at e.value (vendors~main.b46266fa.js:1)
    at vendors~main.b46266fa.js:1
    at Set.forEach (<anonymous>)
    at Object.next (vendors~main.b46266fa.js:1)
    at _ (vendors~main.b46266fa.js:1)
    at w (vendors~main.b46266fa.js:1)
```

**To Reproduce**
Follow the steps Setting up the marketplace ie:

1. Deploy @ 0x01
2. Deploy @ 0x02
3. Execute transaction3 from 0x01
4. Execute transaction4 from 0x02
5. Execute transaction5 from 0x01
6. Deploy @ 0x03
_7. Execute transaction1 from 0x01. <---- point of failure_

**Screenshots**
![image](https://user-images.githubusercontent.com/8539006/88857233-2e780900-d1f6-11ea-8240-a49b9aa05c70.png)

**Desktop (please complete the following information):**
 - OS:  OS X
 - Browser: chrome (84.0.4147.89), brave (V1.11.104)

**Comment by timmymmit** [comment]
@turbolent  to take a look and see if it is reproducible after recent deployment.

**Comment by turbolent** [comment]
@yellowBirdy Thank you for the great bug report! 

The playground was updated and this seems to have fixed the missing type conversion. I just followed the steps you listed and they seem to work and not result in an error.

I'll close this as fixed, but please feel free to comment / reopen if it is still not working for you.

**Comment by rlkelly** [comment]
I believe this should be reopened, I'm getting the same error initially reported

**Comment by danielhour** [comment]
@rlkelly @turbolent @10thfloor I was able to work around this by making a minor update in the Marketplace Contract & in Transaction1.

**TLDR; I pass a reference to the stored Vault vs the Capability itself as a parameter in `createSaleCollection`**
_(I made this change because it looked, code-wise, that the capability was only needed to have `borrow()` be called on it. So instead of calling borrow() in Marketplace.cdc, I call it in Transaction1.cdc)_

______________________________


In `Marketplace.cdc`:
    `access(account) let ownerVault: Capability<&AnyResource{FungibleToken.Receiver}>`
becomes
    `access(account) let ownerRef: &AnyResource{FungibleToken.Receiver}`

and Line 104-108 becomes
    `self.ownerRef.deposit(from: <-buyTokens)`

and 
    `pub fun createSaleCollection(ownerVault: Capability<&AnyResource{FungibleToken.Receiver}>): @SaleCollection`
becomes
    `pub fun createSaleCollection(ownerRef: &AnyResource{FungibleToken.Receiver}): @SaleCollection`

And then in `Transaction1.cdc`:
    `let receiver = acct.getCapability<&{FungibleToken.Receiver}>(/public/MainReceiver)`
becomes
    `let receiver = acct.getCapability(/public/MainReceiver)!.borrow<&FungibleToken.Vault{FungibleToken.Receiver}>()`

______________________________

The Flow team may have a reason for submitting the Capability vs the reference so not sure if this is the correct solution. However, after making this update, I was able to get through the rest of the tutorial without a problem!

If this is ok, should I submit a PR for this?
